cloud_functions/utils.js

```diff
@@ -0,0 +1,31 @@
+const { SecretManagerServiceClient } = require('@google-cloud/secret-manager')
+
+
+// Used to make sure console.error() is given an Error in order to invoke Cloud Logging Error Reporting
+// See https://cloud.google.com/functions/docs/monitoring/error-reporting
+const coerceToError = e => {
+  if (!(e instanceof Error)) {
+    return new Error(`Non-Error thrown: ${e.toString()}`)
+  } else {
+    return e
+  }
+}
+
+// TODO: Share this code with the bard GAE app (will probably require a more sophisticated build/deploy process)
+const getSecret = async ({ project, secretName }) =>  {
+  const client = new SecretManagerServiceClient()
+  const name = `projects/${project}/secrets/${secretName}/versions/latest`
+
+  try {
+    const [version] = await client.accessSecretVersion({ name })
+    return version.payload.data.toString('utf8')
+  } catch (error) {
+    return null
```

Comment:
Do we need to catch here, or can we just let the error propagate out?

Reply:
Interesting question. The context and the way this is used is a little different here in a cloud function than it is in GAE (and I wonder if it could be done differently there, too). Keeping the function exactly the same makes it easier to safely de-duplicate later. On the other hand, I could simplify this case and add to the comment above this, which might encourage improvement on the GAE side when de-duplicating. I think I'll give that a go.